### ISSUE TITLE
MekHQ screen cascading.

### ISSUE THREAD

**Issue by Vincent3g3** [MegaMek/mekhq#125]
Not sure what else to call it but I tried 3.29 and after a few months in with several fights and no issues this began to happen.
![screenshot 2](https://cloud.githubusercontent.com/assets/20602996/17068320/d63d9d42-5015-11e6-8b0a-ce7e2603dfb2.png)

Enlarging or shrinking the screen will fix the issue temporarily. But advancing day or reloading will not.  I downloaded and set up a second 3.29 and got the same issue.

**Comment by Akjosch** [comment]
Does this persist after full restart? Otherwise it looks like some kind of resource leak leading to multiple interstellar map panels being drawn ...

**Comment by pheonixstorm** [comment]
I have not seen this before. Might want to post your HQ log and a zipped campaign + customs if any.

Akjosch, could it be an issue with the theme?

**Comment by Vincent3g3** [comment]
Yes full restart and on any of the pages not just the map. It seems to be most frequently after I have hired someone new or try to add a unit to an ATB generated mission but has happened about anywhere.  It also happens with the three desktop themes I tested as well.  I do not know how to zip a file I will try to figure that out today and add it.


[mekhqlog.txt](https://github.com/MegaMek/mekhq/files/379788/mekhqlog.txt)


![screenshot 5](https://cloud.githubusercontent.com/assets/20602996/17078398/d810f004-50b7-11e6-9d3c-6b4234432b99.png)

**Comment by Vincent3g3** [comment]
The customs are machines I have downloaded from other sources. The homemade are ones I did myself.  I went back and deleted all the non stock machines from my current save but it still had the same issue.  I believe the save marked Tile is the most current one since the issue.

[customs.zip](https://github.com/MegaMek/mekhq/files/379931/customs.zip)
[homemade.zip](https://github.com/MegaMek/mekhq/files/379932/homemade.zip)
[campaigns.zip](https://github.com/MegaMek/mekhq/files/379933/campaigns.zip)

**Comment by Vincent3g3** [comment]
Just to add. I downloaded 3.3 clean and added nothing of my own. No mechs, portraits maps anything and have the same issue. I'm guessing it must be something wrong with my comp.

**Comment by pheonixstorm** [comment]
Possible. If you are running windows run dxdiag. Might need to google it, been a long time since I've had to use it but it should spit out all your driver information and what hardware you are using. Post the file if you can.

A few other things to try would be to reinstall java, check the memory setting in launch4j as well as what java has set aside.

**Comment by Vincent3g3** [comment]
I had uninstalled and reinstalled 64bit java prior to trying mekhq3.30. I also adjusted the launch file at that time to increase memory. I have been unable to figure out how to see what java has set aside. I did notice several mekhq related errors at the bottom of the Dxdiag but I have no idea what they mean.  Google keeps leading me to regedit which seems to be beyond my ability to navigate. I ran the Dxdiag prior to running mekhq if that matters.

[DxDiag.txt](https://github.com/MegaMek/mekhq/files/381876/DxDiag.txt)

**Comment by pheonixstorm** [comment]
dxdiag is the DirectX diagnostic tool. It doesn't really help for java issues but it does give the developer (not normally done in this case) complete information about your hardware, driver version, and some other stuff I dont recall. 

regedit is the default windows registry editor and not likely something you will ever need to touch (curiosity is NOT an excuse to use it unless you want to risk screwing up your computer)

Java memory usage... not sure, I tried looking for it unde java 8 but didn't find it. I know it is there for Java 7 but I don't recall how to get to it. You can try Start -> All Apps -> Java -> configure java it might be there for you.

Looked over the dxdiag and didn't notice anything that stands out. Overall system memory is good, video memory is good, CPU speed is good.... May need to find if the issue is between java and the ATI card.  My system specs are slightly higher than yours, but the main difference is my video is Nvidia and not ATI. I know from past experience when having graphics issues it mainly has to do with the drivers fixing things, but that is mostly for 3D games so my above statement may be totally wrong in this case.

**Comment by Vincent3g3** [comment]
I appreciate you  looking this over for me. At least now I know I won't have missed something glaringly obvious.  I have found some people complaining about my card and java not working together well on other games but their fix involves adding lines of code that seems pretty game issue specific. I guess I will just have to wait for my next computer to try this game again. Thank you for the help.

**Comment by pheonixstorm** [comment]
Don't give up so easily yet. Point me toward these other game issues as well as the java versions they mention. Also what java version are you using? It's possible there is a known bug for either that has some workaroud.

**Comment by Vincent3g3** [comment]
3.16 was the last time I had a version that I did not have to give up on (for completely different bugs). But I keep reading about something called immersive shell being an issue. That however is on Minecraft vs AMI cards.  I also found

http://forum.worldofwarships.com/index.php?/topic/42136-weird-graphical-glitch-with-windows-10/

Which looked kind of similar but  the man who answered the problem said 

"You need to put your CCC settings on AMD enhanced or Application controlled.
Like with many games you can't attempt to override certain game optimizations.  When you installed your drivers it changed a setting for some silly reason."


Which I'll be damned if I can figure out how to do .

**Comment by pheonixstorm** [comment]
Some light reading 
https://support.trionworlds.com/hc/en-us/articles/203374338-How-to-set-up-your-Nvidia-Control-Panel-ATI-Catalyst-Control-Center-to-use-your-game-s-default-settings
http://www.tomsguide.com/us/catalyst-control-center,review-2012.html

If I had access to an ATI rig I could probably be of more help but I've bought Nvidia gear since 94, same with AMD. I will see if I can find a better how-to if the first link doesn't work. The second link is just some boring tech info on the control center.

**Comment by Vincent3g3** [comment]
Well I did discover that the reason I cannot find the CCC is that the newer versions look nothing like those pictures. Sadly you can no longer turn off anti-aliasing only leave it as do what the program says. I did get it to recognize that Mekhq was a game though. I deleted and re downloaded everything to do with my video card I could find. I also tried every single combination  but nothing seemed to have any effect.   What I have a hard time understanding is I had this same problem with 3.27. I assumed it was  a bug and just decided to wait for a newer version. I get 3.29 and play for several months of game time and multiple battles with no issue. Now even the clean 3.30 takes only a few minutes to get it to foul up.  I just keep wondering if it was a graphic card issue why did it take so long to kick in. I would assume it was a new driver had I not had the same problem with 3.27.

**Comment by pheonixstorm** [comment]
What version of Java are you using?

It is either something not interacting correctly with Java or your drivers. Support for non DirectX stuff has never been a high priority on windows.

Last item to try. Uninstall ALL of your graphics drivers, delete the folder thats left, reboot and do a clean install. You may want to try the same with your Java install. This has cleared up past graphics errors for OpenGL (and to some extent DirectX) issues with several (commercial) 3D games. It's a lot to attempt but if you have issues outside of MekHQ this could help clear them up. Up to you, only other option is to go buy a GeForce card lol

On a related note, don't buy a rig with an integrated Intel graphics board. You think this issue is bad.. OpenGL support on the Intel line is abysmal.

**Comment by Vincent3g3** [comment]
Java version is Version 8 Update 101 build 1.8.0_101-b13

I have deleted java/drives previously but not at the same time. I guess I will do that and try to figure out how to load an older version of java. I have not had any issues with any other games but I do not have many.. In fact I do not think I have any other Java games.  I think I'll try to find one to download and test. Fallout 4 and Skyrim are about the only modern games I own. I have world of Xeen on the comp is how out of date I am.

**Comment by pheonixstorm** [comment]
I have Java 8 update 77 installed. Makes me wonder if I should update or not... My most recent game is the Witcher III I got free when I bought my GeForce 960 card.

**Comment by Vincent3g3** [comment]
Well I thought we had it for a minute. I used a driver sweeper program and discovered that somehow I had Nvidia Physx and Vulkan runtime on this comp from a few months ago. But it only ran smooth for about 40 minutes before reverting to the same problem.  Now it is just like before only take s a few seconds to get it to fail. I did download a free java game and had no issues with it. Just for some reason the right click sets it off. I never could figure out how to roll back the java versions to an older one.

**Comment by Vincent3g3** [comment]
Finally. I decided to try to look from the right click angle since that seems to precede the problem.  I ended up with some shellex viewer program because windows 10 seems to half lots of right click issues and found a couple of AMD programs that were set up away from the rest and marked with numbers. Anyway delete them and played about an hour and a half with no issue.  Thank you very much for the help and my next computer is Nvidia all the way.
